File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> ht;
        
        vector<int> temp;
        
        for(int i = 0; i < nums.size(); i++) {
            if(!ht.count(target-nums[i])) {
                ht[nums[i]] = i;
            } else {
                temp = {ht[target-nums[i]], i};
                break;
            }
        }
        
        return temp;
        
    }
};

File: 1-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> hm = new HashMap<Integer, Integer>();
        
        int[] temp = new int[2];
        
        for(int i = 0; i < nums.length; i++) {
            if(hm.containsKey(target-nums[i])) {
                temp[0] = hm.get(target-nums[i]);
                temp[1] = i;
            } else {
                hm.put(nums[i], i);
            }
        }
        
        return temp;
        
    }
}